===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/aenc/Makefile.am,v
retrieving revision 1.17
retrieving revision 1.18
diff -u -r1.17 -r1.18
--- mjpeg/mjpeg_play/aenc/Makefile.am	2003/11/25 10:36:30	1.17
+++ mjpeg/mjpeg_play/aenc/Makefile.am	2004/03/03 17:55:45	1.18
@@ -27,4 +27,4 @@
 	table_th.h \
 	wav_io.h
 
-mp2enc_LDADD = $(top_builddir)/utils/libmjpegutils.a $(LIBM_LIBS)
+mp2enc_LDADD = $(top_builddir)/utils/libmjpegutils.la $(LIBM_LIBS)
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/lavtools/Makefile.am,v
retrieving revision 1.90
retrieving revision 1.92
diff -u -r1.90 -r1.92
--- mjpeg/mjpeg_play/lavtools/Makefile.am	2004/01/01 19:59:55	1.90
+++ mjpeg/mjpeg_play/lavtools/Makefile.am	2004/03/03 17:55:45	1.92
@@ -8,6 +8,8 @@
         -I /usr/X11R6/include -I$(top_srcdir)/utils \
 	${GTK_CFLAGS} $(JPEG_CFLAGS)
 
+LIBMJPEGUTILS = $(top_builddir)/utils/libmjpegutils.la
+
 # *********************************************************************
 # Support lib
 
@@ -133,113 +136,108 @@
 	yuvcorrect.h
 
 ypipe_SOURCES = ypipe.c
-ypipe_LDADD = $(top_builddir)/utils/libmjpegutils.a
+ypipe_LDADD = $(LIBMJPEGUTILS)
 
 yuv2lav_SOURCES = yuv2lav.c
-yuv2lav_LDADD = $(top_builddir)/utils/libmjpegutils.a \
-		 liblavfile.la liblavjpeg.la ${LIBM_LIBS}
+yuv2lav_LDADD = $(LIBMJPEGUTILS) liblavfile.la liblavjpeg.la ${LIBM_LIBS}
 
 transist_flt_SOURCES = transist.flt.c
-transist_flt_LDADD = $(top_builddir)/utils/libmjpegutils.a
+transist_flt_LDADD = $(LIBMJPEGUTILS)
 
 matteblend_flt_SOURCES = matteblend.flt.c
-matteblend_flt_LDADD = $(top_builddir)/utils/libmjpegutils.a
+matteblend_flt_LDADD = $(LIBMJPEGUTILS)
 
 multiblend_flt_SOURCES = multiblend.flt.c
-multiblend_flt_LDADD = $(top_builddir)/utils/libmjpegutils.a
+multiblend_flt_LDADD = $(LIBMJPEGUTILS)
 
 if HAVE_X
   LAVPLAY_X_LIBS = $(X_LIBS) -lX11
 endif
 
 lavplay_SOURCES = lavplay.c
-lavplay_LDADD = $(top_builddir)/utils/libmjpegutils.a \
+lavplay_LDADD = $(LIBMJPEGUTILS) \
 	liblavplay.la liblavjpeg.la liblavfile.la \
 	@LIBGETOPT_LIB@ ${PTHREAD_LIBS} ${SDL_LIBS} $(LAVPLAY_X_LIBS)
 
 lavrec_SOURCES = lavrec.c
-lavrec_LDADD = $(top_builddir)/utils/libmjpegutils.a \
+lavrec_LDADD = $(LIBMJPEGUTILS) \
 	liblavrec.la liblavfile.la liblavjpeg.la \
 	@LIBGETOPT_LIB@ $(PTHREAD_LIBS)
 
 testrec_SOURCES = testrec.c audiolib.c
-testrec_LDADD = $(PTHREAD_LIBS) $(top_builddir)/utils/libmjpegutils.a
+testrec_LDADD = $(PTHREAD_LIBS) $(LIBMJPEGUTILS)
 
 lav2yuv_SOURCES = lav2yuv.c lav_common.c
-lav2yuv_LDADD = $(top_builddir)/utils/libmjpegutils.a \
+lav2yuv_LDADD = $(LIBMJPEGUTILS) \
 		liblavfile.la liblavjpeg.la $(DV_LIBS)
 
 jpeg2yuv_SOURCES = jpeg2yuv.c
-jpeg2yuv_LDADD = $(top_builddir)/utils/libmjpegutils.a liblavjpeg.la
+jpeg2yuv_LDADD = $(LIBMJPEGUTILS) liblavjpeg.la
 
 png2yuv_SOURCES = png2yuv.c colorspace.c subsample.c
-png2yuv_LDADD = $(top_builddir)/utils/libmjpegutils.a -lpng -lz -lm
+png2yuv_LDADD = $(LIBMJPEGUTILS) -lpng -lz -lm
 
 lavpipe_SOURCES = lavpipe.c pipelist.c
-lavpipe_LDADD = $(top_builddir)/utils/libmjpegutils.a
+lavpipe_LDADD = $(LIBMJPEGUTILS)
 
 yuvmedianfilter_SOURCES = yuvmedianfilter.c
-yuvmedianfilter_LDADD = $(top_builddir)/utils/libmjpegutils.a
+yuvmedianfilter_LDADD = $(LIBMJPEGUTILS)
 
 lav2wav_SOURCES = lav2wav.c
-lav2wav_LDADD = $(top_builddir)/utils/libmjpegutils.a \
-	 liblavfile.la
+lav2wav_LDADD = $(LIBMJPEGUTILS) liblavfile.la
 
 lavtrans_SOURCES = lavtrans.c
-lavtrans_LDADD = $(top_builddir)/utils/libmjpegutils.a \
-	 liblavfile.la
+lavtrans_LDADD = $(LIBMJPEGUTILS) liblavfile.la
 
 lavaddwav_SOURCES = lavaddwav.c
-lavaddwav_LDADD = $(top_builddir)/utils/libmjpegutils.a \
-	 liblavfile.la
+lavaddwav_LDADD = $(LIBMJPEGUTILS) liblavfile.la
 
 glav_SOURCES = glav.c glav_main.c
-glav_LDADD = $(top_builddir)/utils/libmjpegutils.a ${GTK_LIBS}
+glav_LDADD = $(LIBMJPEGUTILS) ${GTK_LIBS}
 
 lavvideo_SOURCES = lavvideo.c frequencies.c
 
 yuvscaler_SOURCES = yuvscaler.c yuvscaler_resample.c yuvscaler_bicubic.c
-yuvscaler_LDADD = $(top_builddir)/utils/libmjpegutils.a $(LIBM_LIBS)
+yuvscaler_LDADD = $(LIBMJPEGUTILS) $(LIBM_LIBS)
 
 yuvcorrect_SOURCES = yuvcorrect.c yuvcorrect_functions.c
-yuvcorrect_LDADD = $(top_builddir)/utils/libmjpegutils.a $(LIBM_LIBS)
+yuvcorrect_LDADD = $(LIBMJPEGUTILS) $(LIBM_LIBS)
 
 yuvcorrect_tune_SOURCES = yuvcorrect_tune.c yuvcorrect_functions.c
-yuvcorrect_tune_LDADD = $(top_builddir)/utils/libmjpegutils.a $(LIBM_LIBS)
+yuvcorrect_tune_LDADD = $(LIBMJPEGUTILS) $(LIBM_LIBS)
 
 yuvplay_SOURCES = yuvplay.c
-yuvplay_LDADD = $(SDL_LIBS) $(PTHREAD_LIBS) $(top_builddir)/utils/libmjpegutils.a
+yuvplay_LDADD = $(SDL_LIBS) $(PTHREAD_LIBS) $(LIBMJPEGUTILS)
 
 ppmtoy4m_SOURCES = ppmtoy4m.c colorspace.c subsample.c
-ppmtoy4m_LDADD = $(top_builddir)/utils/libmjpegutils.a
+ppmtoy4m_LDADD = $(LIBMJPEGUTILS)
 
 pgmtoy4m_SOURCES = pgmtoy4m.c
-pgmtoy4m_LDADD = $(top_builddir)/utils/libmjpegutils.a
+pgmtoy4m_LDADD = $(LIBMJPEGUTILS)
 
 y4mtoppm_SOURCES = y4mtoppm.c colorspace.c subsample.c
-y4mtoppm_LDADD = $(top_builddir)/utils/libmjpegutils.a
+y4mtoppm_LDADD = $(LIBMJPEGUTILS)
 
 y4mcolorbars_SOURCES = y4mcolorbars.c colorspace.c subsample.c
-y4mcolorbars_LDADD = $(top_builddir)/utils/libmjpegutils.a
+y4mcolorbars_LDADD = $(LIBMJPEGUTILS)
 
 lavinfo_SOURCES = lavinfo.c
-lavinfo_LDADD = $(top_builddir)/utils/libmjpegutils.a \
-	liblavfile.la
+lavinfo_LDADD = $(LIBMJPEGUTILS) liblavfile.la
 
 y4mshift_SOURCES = y4mshift.c
-y4mshift_LDADD = $(top_builddir)/utils/libmjpegutils.a
+y4mshift_LDADD = $(LIBMJPEGUTILS)
 
 y4mblack_SOURCES = y4mblack.c
-y4mblack_LDADD = $(top_builddir)/utils/libmjpegutils.a
+y4mblack_LDADD = $(LIBMJPEGUTILS)
 
 y4mtoyuv_SOURCES = y4mtoyuv.c
-y4mtoyuv_LDADD = $(top_builddir)/utils/libmjpegutils.a
+y4mtoyuv_LDADD = $(LIBMJPEGUTILS)
 
 yuv4mpeg_SOURCES = yuv4mpeg.c
-yuv4mpeg_LDADD = $(top_builddir)/utils/libmjpegutils.a
+yuv4mpeg_LDADD = $(LIBMJPEGUTILS)
 
 yuvfps_SOURCES = yuvfps.c
-yuvfps_LDADD = ${top_builddir}/utils/libmjpegutils.a
+yuvfps_LDADD = $(LIBMJPEGUTILS)
 
 yuvinactive_SOURCES = yuvinactive.c
-yuvinactive_LDADD = ${top_builddir}/utils/libmjpegutils.a
+yuvinactive_LDADD = $(LIBMJPEGUTILS)
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/mpeg2enc/Makefile.am,v
retrieving revision 1.67
retrieving revision 1.68
diff -u -r1.67 -r1.68
--- mjpeg/mjpeg_play/mpeg2enc/Makefile.am	2003/12/20 17:33:38	1.67
+++ mjpeg/mjpeg_play/mpeg2enc/Makefile.am	2004/03/03 17:55:46	1.68
@@ -26,6 +26,8 @@
 
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/utils
 
+LIBMJPEGUTILS = $(top_builddir)/utils/libmjpegutils.la
+
 lib_LTLIBRARIES = libmpeg2encpp.la
 
 if HAVE_ASM_MMX
@@ -113,24 +115,24 @@
 	-export-dynamic 
 
 libmpeg2encpp_la_DEPENDENCIES = \
-	$(top_builddir)/utils/libmjpegutils.a \
+	$(LIBMJPEGUTILS) \
 	$(top_builddir)/utils/libcpuaccel.la \
 	$(top_builddir)/utils/libmotion.la \
 	$(LIBSIMDASM_LIB)
 
 libmpeg2encpp_la_LIBADD = \
-	$(top_builddir)/utils/libmjpegutils.a \
+	$(LIBMJPEGUTILS) \
 	$(top_builddir)/utils/libcpuaccel.la \
 	$(top_builddir)/utils/libmotion.la \
 	$(LIBSIMDASM_LIB)
 
 mpeg2enc_DEPENDENCIES = \
-	$(top_builddir)/utils/libmjpegutils.a \
+	$(LIBMJPEGUTILS) \
 	$(top_builddir)/mpeg2enc/libmpeg2encpp.la
 
 mpeg2enc_LDADD = \
 	$(top_builddir)/mpeg2enc/libmpeg2encpp.la \
-	$(top_builddir)/utils/libmjpegutils.a \
+	$(LIBMJPEGUTILS) \
 	@PTHREAD_LIBS@ @LIBGETOPT_LIB@ $(LIBM_LIBS)
 
 # VITAL: Override default compilation rules generated by automake...
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/mplex/Makefile.am,v
retrieving revision 1.41
retrieving revision 1.42
diff -u -r1.41 -r1.42
--- mjpeg/mjpeg_play/mplex/Makefile.am	2003/12/20 17:33:38	1.41
+++ mjpeg/mjpeg_play/mplex/Makefile.am	2004/03/03 17:55:46	1.42
@@ -67,7 +67,7 @@
 	-export-dynamic 
 
 libmplex2_la_LIBADD = \
-	$(top_builddir)/utils/libmjpegutils.a
+	$(top_builddir)/utils/libmjpegutils.la
 
 libmplex2_la_CXXFLAGS = $(ZALPHA_FLAGS)
 
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/yuvdenoise/Makefile.am,v
retrieving revision 1.18
retrieving revision 1.19
diff -u -r1.18 -r1.19
--- mjpeg/mjpeg_play/yuvdenoise/Makefile.am	2003/12/20 17:33:39	1.18
+++ mjpeg/mjpeg_play/yuvdenoise/Makefile.am	2004/03/03 17:55:46	1.19
@@ -25,5 +25,5 @@
 
 yuvdenoise_LDADD = \
 	$(top_builddir)/utils/libcpuaccel.la \
-	$(top_builddir)/utils/libmjpegutils.a \
+	$(top_builddir)/utils/libmjpegutils.la \
 	@LIBM_LIBS@
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/yuvfilters/Makefile.am,v
retrieving revision 1.20
retrieving revision 1.21
diff -u -r1.20 -r1.21
--- mjpeg/mjpeg_play/yuvfilters/Makefile.am	2003/12/21 14:37:44	1.20
+++ mjpeg/mjpeg_play/yuvfilters/Makefile.am	2004/03/03 17:55:46	1.21
@@ -7,7 +7,7 @@
 noinst_LTLIBRARIES = libyuvfilters.la
 
 AM_CPPFLAGS = -I$(top_srcdir)/utils
-MJPEGLIB = $(top_builddir)/utils/libmjpegutils.a
+MJPEGLIB = $(top_builddir)/utils/libmjpegutils.la
 
 libyuvfilters_la_SOURCES = \
 	addtask.c \
===================================================================
RCS file: /cvsroot/mjpeg/mjpeg_play/utils/Makefile.am,v
retrieving revision 1.21
retrieving revision 1.22
diff -u -r1.21 -r1.22
--- mjpeg/mjpeg_play/utils/Makefile.am	2003/12/20 17:33:38	1.21
+++ mjpeg/mjpeg_play/utils/Makefile.am	2004/03/03 17:55:46	1.22
@@ -15,10 +15,14 @@
 
 noinst_LTLIBRARIES = libmotion.la libcpuaccel.la
 
-# FIXME: make this libtool (but static-only!) too
-lib_LIBRARIES = libmjpegutils.a
+lib_LTLIBRARIES = libmjpegutils.la
 
-libmjpegutils_a_SOURCES = \
+libmjpegutils_la_LDFLAGS = \
+	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
+	-release $(LT_RELEASE) \
+	-export-dynamic
+
+libmjpegutils_la_SOURCES = \
 	mjpeg_logging.c \
 	mpegconsts.c \
 	mpegtimecode.c \
